src/ferc_xbrl_extractor/instance.py

```diff
@@ -321,13 +323,11 @@ def parse(self, fact_prefix: str = "ferc") -> Instance:
         parser = etree.XMLParser(huge_tree=True)
 
         # Check if instance contains path to file or file data and parse accordingly
-        if isinstance(self.file, str):
+        try:
             tree = etree.parse(self.file, parser=parser)  # nosec: B320
-            root = tree.getroot()
-        elif isinstance(self.file, io.BytesIO):
```

Comment:
An `io.BytesIO` is "file-like" so we can just shove it into `etree.parse()` regardless.